nupic/embodied/agents/curious_ppo_agent.py

```diff
@@ -0,0 +1,503 @@
+        nsteps_per_seg,
+        nsegs_per_env,
+        expName,
+        vLogFreq,
+        dynamics_list,
+    ):
+        self.dynamics_list = dynamics_list
+        self.n_updates = 0
+        self.scope = scope
+        self.device = device
+        self.ob_space = ob_space
+        self.ac_space = ac_space
+        self.stochpol = stochpol
+        self.nepochs = nepochs
+        self.lr = lr
+        self.cliprange = cliprange
+        self.nsteps_per_seg = nsteps_per_seg
+        self.nsegs_per_env = nsegs_per_env
+        self.nminibatches = nminibatches
+        self.gamma = gamma
+        self.lam = lam
+        self.normrew = normrew
+        self.normadv = normadv
+        self.use_news = use_news
+        self.ent_coef = ent_coef
+        self.ext_coeff = ext_coeff
+        self.int_coeff = int_coeff
+        self.vLogFreq = vLogFreq
+        # TODO: Add saving with expName & vLog Freq video saving
+
+    def start_interaction(self, env_fns, dynamics_list, nlump=1):
+        """Set up environments and initialize everything.
+
+        Parameters
+        ----------
+        env_fns : [envs]
+            List of environments, optionally with wrappers etc.
+        dynamics_list : [Dynamics]
+            List of dynamics models.
+        nlump : int
+            ..
+
+        """
+        # Specify parameters that should be optimized
+        param_list = (
+            self.stochpol.param_list + self.dynamics_list[0].auxiliary_task.param_list
+        )
+        for dynamic in self.dynamics_list:
+            param_list = param_list + dynamic.param_list
+
+        # Initialize the optimizer
+        self.optimizer = torch.optim.Adam(param_list, lr=self.lr)
+        # Set gradients to zero
+        self.optimizer.zero_grad()
+
+        # set parameters
+        self.nenvs = nenvs = len(env_fns)
+        self.nlump = nlump
+        self.lump_stride = nenvs // self.nlump
+        # Initialize list of VecEnvs
+        self.envs = [
+            VecEnv(
+                env_fns[lump * self.lump_stride : (lump + 1) * self.lump_stride],
+                spaces=[self.ob_space, self.ac_space],
+            )
+            for lump in range(self.nlump)
+        ]
+
+        # Initialize rollout class for experience collection
+        self.rollout = Rollout(
+            ob_space=self.ob_space,
+            ac_space=self.ac_space,
+            nenvs=nenvs,
+            nsteps_per_seg=self.nsteps_per_seg,
+            nsegs_per_env=self.nsegs_per_env,
+            nlumps=self.nlump,
+            envs=self.envs,
+            policy=self.stochpol,
+            int_rew_coeff=self.int_coeff,
+            ext_rew_coeff=self.ext_coeff,
+            dynamics_list=dynamics_list,
+        )
+
+        # Initialize buffers for advantages and returns of each rollout
+        self.buf_advs = np.zeros((nenvs, self.rollout.nsteps), np.float32)
+        self.buf_rets = np.zeros((nenvs, self.rollout.nsteps), np.float32)
+
+        if self.normrew:
```

Comment:
There are too many variable abbreviations used, which makes the code hard to read for someone unfamiliar with it. 

Abbreviations are generally ok if the meaning is obvious, but not all of them are. For example we could change `normrew` to `normalize_rewards`, or ent_coef to `entropy_coef`or ext_coeff to `external_reward_coef`. The code should preferably read like a paper. That is going to help not only reading the code, and associating it with the method described in the paper, but also maintaining the code (when we need to search for a variable for example).